### ISSUE TITLE
Change default pkg-config

### ISSUE THREAD

**Issue by madebr** [mesonbuild/meson#7701]
On Windows, I use `pkgconf` as `pkg-config` implementation.
Its executable is called `pkgconf`.
Finding a dependency that provides a .pc fails with the following output:
```
Did not find pkg-config by name 'pkgconfig'
Found Pkg-config: NO
```
Then it goes on trying cmake...

I've set the `PKG_CONFIG` environment variable,
but that doesn't work because meson doesn't do environment variables.

The reason is that meson only looks for a limited number of pkg-config names:
https://github.com/mesonbuild/meson/blob/104b80a75c1a092ab70feb65fb1ef1c592d45870/mesonbuild/dependencies/base.py#L580-L607
The default is `pkg-config`.

How can I add `pkgconf` to this list or change the default?

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)? NO
* what operating system **Windows 10**
* what Python version are you using e.g. 3.8.0: **3.7.7**
* what `meson --version` **0.55.0**
* what `ninja --version` if it's a Ninja build **1.10.0**

**Comment by madebr** [comment]
Ok, I found the culprit.
Setting:
```
PKG_CONFIG=C:/Users/maarten/.conan/data/pkgconf/1.7.3/_/_/package/6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7/bin/pkgconf
```
in my environment does **not** work, but
```
PKG_CONFIG=C:/Users/maarten/.conan/data/pkgconf/1.7.3/_/_/package/6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7/bin/pkgconf.exe
```
does.

The `.exe` suffix is required.

**Comment by orbea** [comment]
I agree that meson should check if `pkgconf` exists first and then check for other names. If the user has `pkgconf` installed that is likely the preferred choice.

**Comment by nirbheek** [comment]
I don't think we should do this. pkgconf provides a pkg-config wrapper already, and meson also provides mechanisms for picking the correct binary.

> The `.exe` suffix is required.

This is a bug and should be fixed. [We have some tests for this already](https://github.com/mesonbuild/meson/blob/88e13c5f7c8e977ed879cabaa2800c211a536b60/run_unittests.py#L5476-L5479), so it should be working. Need to investigate why it isn't.

**Comment by orbea** [comment]
> I don't think we should do this. pkgconf provides a pkg-config wrapper already

The pkg-config symlink is entirely downstream and is not provided by pkgconf. Its merely there for backwards compatibility and may vanish sometime in the future.

> If you want pkgconf to be used when you invoke pkg-config, you should install a symlink for this. We do not do this for you, as we believe it is better for vendors to make this determination themselves.

https://github.com/pkgconf/pkgconf#pkg-config-symlink

**Comment by xclaesse** [comment]
We have Environment.default_pkgconfig, I think we should add pkgconf in that list.

**Comment by nirbheek** [comment]
> We have Environment.default_pkgconfig, I think we should add pkgconf in that list.

True, we should do that. Not sure if `pkgconf` should be higher priority than `pkg-config`, though. I guess we can add it as `['pkg-config', 'pkgconf']` to be safer for now and flip it at some point.

**Comment by brunolopesdsilv** [comment]
>  Not sure if pkgconf should be higher priority than pkg-config

Arch, Fedora, Debian(?) and MSYS2 actually use **pkgconf** as default, but meson continues calling the pkg-config symlink...

**Comment by zhaob1n** [comment]
meson still doesn't look for pkgconf, any reasons?

**Comment by bonzini** [comment]
Mostly that pkg-config works; pull requests are welcome (if you open one, add a release notes snippet in docs/markdown/snippets - search the git history for examples).
